### ISSUE TITLE
Start-Process -Credential prevents interactive input in both the caller and the new window

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#12129]
This is a long-standing bug that also affects Windows PowerShell.

On Windows, `-Credential` can be used to start a process in a new window with a given user identity.

Currently, doing so effectively prevents interactive use of both the caller's window and the new window, because they seem to be competing for keyboard input.

Note: Curiously, if you start `powershell.exe` specifically, the problem does _not_ occur; other shells, however - such as `pwsh` itself and `cmd` - do exhibit the problem.

_Update_:

 * The problem may be related to the `PSReadLine` module, given that unloading it first (`Remove-Module PSReadLine`) makes the problem go away - see https://github.com/PowerShell/PSReadLine/issues/288

# Steps to reproduce

On Windows (_update_: problem verified on Windows 10 versions 1809 and 1909 - see `(Get-ComputerInfo).WindowsVersion`)

```powershell
# Enter a different user's credentials (or even your own) when prompted.
Set-Location C:\; Start-Process cmd -Credential (Get-Credential)
```

# Expected behavior

The call should return after having launched the `cmd` process and both the calling shell and the `cmd` shell in the new window should function normally and independently.

# Actual behavior

Typing in either window either has no effect or sometimes makes what was typed show up in the respective other window.

Behavior returns to normal when the new window is closed.

# Environment data

```none
PowerShell Core v7.0.0 (7.0.0) on Microsoft Windows 10 Pro (64-bit; Version 1909, OS Build: 18363.720)
```

# Workarounds

* Use `runas.exe`

* Launch the initial process hidden, non-interactively, and use it as an intermediary to launch an interactive session in a new, visible window; e.g.:

```powershell
Start-Process -WindowStyle Hidden -Credential (Get-Credential) cmd '/c start cmd' -WorkingDirectory c:\
```

**Comment by iSazonov** [comment]
I can not repo. Run pwsh, run the command, new window is opened, both windows work well.
Could you please add more info about repo steps? Do you mean using `-NoNewWindow`?

**Comment by mklement0** [comment]
I just tried the repro steps again - I get the symptom every time, in the environment as reported.
Are you running on a different Windows 10 version?

`-NoNewWindow` doesn't apply, because it is ignored when you use `-Credential` (the combination should arguably result in a syntax error, but that's a separate issue).

**Comment by iSazonov** [comment]
```powershell
Name                           Value
----                           -----
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iSazonov** [comment]
I tried on 10.0.19041 and I can repo.

**Comment by iSazonov** [comment]
Look like defaults for redirections was changed on OS level.

**Comment by iSazonov** [comment]
@mklement0 Have you Windows Feedback tool on your system? You could pull the feedback there (and maybe reference the issue there) and share a reference here (I'd confirm your report).

/cc @TravisEz13 @PaulHigin Could you please look the issue?

**Comment by mklement0** [comment]
@iSazonov, since `runas.exe /user:...` still manages to work correctly on the affected recent Windows 10 versions, I suspect that a solution on our end is possible.

**Comment by iSazonov** [comment]
@mklement0 I hope but we use so simple pinvoke that I wonder where we have an issue.

**Comment by PaulHigin** [comment]
+ @daxian-dbw 
I believe Dongbo has worked most recently in this area, and may have some thoughts.

**Comment by gerardog** [comment]
I can repro this on all my machines with Win11 Pro 10.0.22000.1098. See #18333
Pwsh v7.2.6, using WT or ConHost.
Not ocurring when running ` Start-Process cmd -Credential (Get-Credential)` from windows powershell.

**Comment by mklement0** [comment]
That is curious, @gerardog - on my W11 machine I see it in Windows PowerShell too, both in regular console windows and in Windows Terminal.

However, when I run `Remove-Module PSReadLine` first, it starts working - in _both_ PS editions (obviously, that's not a practical solution).

**Comment by SteveL-MSFT** [comment]
The @PowerShell/wg-powershell-cmdlets repro'd this and agree that this is a bug in PSReadLine

**Comment by jamie-sandbox** [comment]
## Workaround

https://github.com/PowerShell/PSReadLine/issues/288#issuecomment-1837606501

**Comment by mklement0** [comment]
It turned out to be a PowerShell problem after all, and was fixed in #20853 - it isn't clear yet in what version the fix will land.
